examples/dune

```diff
@@ -0,0 +1,5 @@
+(executable
+  (name         mirage_prometheus_example)
+  (package      prometheus)
+  (public_name  mirage-prometheus-example)
```

Comment:
We probably don't want `public_name` here. There's not much point in installing this binary to the user's path.

But even if we do, there's no need to rename the source file too - `name` doesn't need to match `public_name`.

Reply:
What you describe is what I attempted (and failed) to do. I've probably misunderstood some part of this.

Reply:
```
File "examples/dune", line 3, characters 2-27:
3 |   (package      prometheus)
      ^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This field is useless without a (public_name ...) field.
```

(This, I believe, is what led me to add `public_name`.

Removing `(package      prometheus)` 

```
File "examples/dune", line 2, characters 16-41:
2 |   (name         mirage_prometheus_example)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Module "Mirage_prometheus_example" doesn't exist.
```

![image](https://user-images.githubusercontent.com/211/65830965-1836af00-e2ac-11e9-8957-1fb1342e412a.png)

Could you tell me where to step from there?

Reply:
I'll punt to having the name `examples/example.ml`.

Reply:
Force-pushed a more minimal change, avoiding the install step.